src/mqttcm_connect.c

```diff
@@ -1082,10 +1124,20 @@ rbusError_t MqttSubscribeSetHandler(rbusHandle_t handle, rbusProperty_t prop, rb
 					}
 					subscribe = strdup(data);
 					free(data);
+					if(clientId !=NULL)
+					{
+						snprintf(topic,MAX_MQTT_LEN,"%s%s", MQTT_SUBSCRIBE_TOPIC_PREFIX,clientId);
+						MqttCMInfo("The value of topic is %s\n",topic);
+					}
 					MqttCMInfo("mqtt subscribe %s\n", subscribe);
-					mqtt_subscribe();
+					if(topic !=NULL)
+						mqtt_subscribe(topic);
```

Comment:
As this is generic code for multiple components, move it outside webconfig check